Tenants: add create/update methods and maxUnits configuration

nicoloboschi opened the pull request:
* Added new endpoint for creating (POST) and updating a tenant (PATCH), similar to the application ones
* The PUT tenant is still there

The new commands can pass a new optional param `--max-units` (integer) (part of https://github.com/LangStream/langstream/issues/304)

The PUT endpoint doesn't accept any body. To create or update the tenant is recommended to use create/update from now on.

eolivelli commented on langstream-api/src/main/java/ai/langstream/api/webservice/tenant/CreateTenantRequest.java:
```diff
@@ -0,0 +1,15 @@
+@AllArgsConstructor
+public class CreateTenantRequest {
+
+    private Integer maxUnits;
```
maxTotalResourceUnits

eolivelli commented on langstream-api/src/main/java/ai/langstream/api/webservice/tenant/TenantConfiguration.java:
```diff
@@ -26,4 +26,5 @@
 @AllArgsConstructor
 public class TenantConfiguration {
     private String name;
+    private int maxUnits;
```
maxTotalResourceUnits

eolivelli commented on langstream-api/src/main/java/ai/langstream/api/webservice/tenant/UpdateTenantRequest.java:
```diff
@@ -0,0 +1,15 @@
+@AllArgsConstructor
+public class UpdateTenantRequest {
+
+    private Integer maxUnits;
```
maxTotalResourceUnits

eolivelli commented on langstream-cli/src/main/java/ai/langstream/cli/commands/tenants/CreateTenantCmd.java:
```diff
@@ -0,0 +1,53 @@
+    private String name;
+
+    @CommandLine.Option(
+            names = {"--max-units"},
```
--max-total-resource-units

eolivelli commented on langstream-cli/src/main/java/ai/langstream/cli/commands/tenants/UpdateTenantCmd.java:
```diff
@@ -0,0 +1,53 @@
+    private String name;
+
+    @CommandLine.Option(
+            names = {"--max-units"},
```
--max-total-resource-units

eolivelli commented on langstream-webservice/src/main/java/ai/langstream/webservice/common/TenantResource.java:
```diff
@@ -62,6 +69,42 @@ TenantConfiguration getTenant(@NotBlank @PathVariable("tenant") String tenant) {
+            @RequestBody CreateTenantRequest createTenantRequest)
+            throws TenantException {
+        try {
+            globalMetadataService.createTenant(tenant, createTenantRequest);
```
we need an hard limit for the "--max-total-resource-units", configurable in the control plane

eolivelli reviewed (changes requested):
nice work!

I have left some comments, PTAL

nicoloboschi commented:
@eolivelli thanks, I renamed the property and added a configurable limit in the control plane config. PTAL again

p.s. I'll do the rename of the deployer/control-plane later

nicoloboschi commented:
Merged as part of https://github.com/LangStream/langstream/pull/326
